transmit/direct_transmit_test.go

```diff
@@ -535,10 +539,26 @@ func TestDirectTransmission(t *testing.T) {
 		allEvents = append(allEvents, event)
 	}
 
-	// Send all events, the stop to make sure everything is fully flushed.
-	for _, event := range allEvents {
+	// Send all but one of our events.
+	for _, event := range allEvents[:len(allEvents)-1] {
 		dt.EnqueueEvent(event)
 	}
+
+	// At this point, time hasn't advanced so we should't get any incomplete batches.
+	require.Eventually(t, func() bool {
+		return len(testServer.getEvents()) == len(allEvents)-6
```

Comment:
Can we add a comment about how we came up with this calculation? 6 includes the dropped event due to size, 4 waiting for timeout, and 1 that hasn't been enqueued